process/smartContract/process.go

```diff
@@ -870,13 +871,15 @@ func (sc *scProcessor) createSCRForSender(
 	scTx.ReturnMessage = []byte(returnMessage)
 	setOriginalTxHash(scTx, txHash, tx)
 
-	scTx.Data = []byte("@" + hex.EncodeToString([]byte(returnCode.String())))
-	for _, retData := range returnData {
-		scTx.Data = append(scTx.Data, []byte("@"+hex.EncodeToString(retData))...)
-	}
-
 	if callType == vmcommon.AsynchronousCall {
 		scTx.CallType = vmcommon.AsynchronousCallBack
+		scTx.Data = []byte("@" + hex.EncodeToString(big.NewInt(int64(returnCode)).Bytes()))
```

Comment:
This change might require a corresponding one in Arwen. Please let me know when you merge this.

Reply:
ok, will try to merge this asap.